### PR TITLE
PILOT-1821: add the Semantic Versioning in upload for code release

### PR DESCRIPTION
**Pull request by colorzzr**
## Summary

add the Semantic Versioning in upload for code release

## JIRA Issues

PILOT-1821

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Are there any new or updated tests to validate the changes?

- [ ] Yes
- [ ] No

## Test Directions

(Additional instructions for how to run tests or validate functionality if not covered by unit tests)

### CODE REVIEW

File: .github/workflows/build-and-publish.yml [hunk 1]
```diff
@@ -45,7 +52,8 @@ jobs:
             ghcr.io/pilotdataplatform/download
           # generate Docker tags based on the following events/attributes
           tags: |
-            type=sha,enable=true,prefix=,suffix=,format=short
+            type=sha,enable=true,prefix=alembic-,suffix=,format=short
```
Comment (andricDu) [inline .github/workflows/build-and-publish.yml:55]: remove the prefix

Comment (colorzzr) [reply .github/workflows/build-and-publish.yml]: removed

File: .github/workflows/build-and-publish.yml [hunk 2]
```diff
@@ -46,6 +57,7 @@ jobs:
           # generate Docker tags based on the following events/attributes
           tags: |
             type=sha,enable=true,prefix=,suffix=,format=short
+            type=raw,prefix=,suffix=,value=${{needs.get_version.outputs.app_version}}, enabled=${{needs.extract-branch-name.outputs.branch=='main'}}
```
Comment (andricDu) [inline .github/workflows/build-and-publish.yml:60]: `enable`

**Review by andricDu — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-publish.yml (hunk 2 above)
Comment (colorzzr) [reply .github/workflows/build-and-publish.yml]: updated

File: .github/workflows/build-and-publish.yml [hunk 3]
```diff
@@ -46,6 +57,7 @@ jobs:
           # generate Docker tags based on the following events/attributes
           tags: |
             type=sha,enable=true,prefix=,suffix=,format=short
+            type=raw,prefix=,suffix=,value=${{needs.get_version.outputs.app_version}}, enable=${{needs.extract-branch-name.outputs.branch=='main'}}
```
Comment (andricDu) [inline .github/workflows/build-and-publish.yml:60]: the job is called `get-version` where previously we were refering to a step called `get_version`

**Review by andricDu — CHANGES_REQUESTED** [review]

**Pull request merged**